src/main/scala/dev/kamu/cli/commands/PullCommand.scala

```diff
@@ -15,20 +16,35 @@ class PullCommand(
   metadataRepository: MetadataRepository,
   sparkRunner: SparkRunner,
   datasetIDs: Seq[String],
-  all: Boolean
+  all: Boolean,
+  recursive: Boolean
 ) extends Command {
   private val logger = LogManager.getLogger(getClass.getName)
 
   def run(): Unit = {
-    val datasets =
-      if (all)
-        metadataRepository.getAllDatasets()
-      else
-        datasetIDs
-          .map(DatasetID)
-          .map(metadataRepository.getDataset)
-
-    val numUpdated = datasets
+    val dependsOn: DatasetID => List[DatasetID] = {
+      if (all || recursive) {
+        metadataRepository.getDataset(_).dependsOn.toList
+      } else {
+        metadataRepository
```

Comment:
Hm, I don't think this is a correct implementation of the non-recursive mode. You should add some tests for it.

Reply:
Not sure how to test that. An example would be useful.